### PR TITLE
*: Move from our fork of sysinfo to upstream and get test suite passing on AMD cpus

### PR DESCRIPTION
**Pull request by nrc**
### What problem does this PR solve?

Currently the TiKV test suite fails on AMD machines with many cores. 

### What is changed and how it works?

This PR fixes that by moving from our fork of sysinfo to upstream, which is a much newer version and includes some fixes that are needed (first four commits).

The fifth commit uses `get_vendor_id` which I added to upstream sysinfo to skip part of a test for AMD CPUs, this is because the structure returned by cpu id on AMD is different to Intel and we can't get that information. 

The sixth commit fixes two tests which consistently fail locally for me. I don't know why `test_check_data_dir` would pass for anyone, as far as I can tell the `data1` subdirectory is never created, so `check_data_dir` should fail, I don't know why it doesn't in CI, etc (older kernel maybe?). `test_debug_fail_point` fails because multiple threads are setting failpoints for different tests (global mutable state :-( ) That means we can't be sure that the failpoints added by the test are the only failpoints that will be listed. I think I am hitting this because with many cores, Cargo runs many tests simultaneously which increases the chance of conflict.

There are also a few typos that I fixed while debugging.

PTAL @lonng @brson 

### Release note

* No release note

### CODE REVIEW

**Comment by nrc** [comment]
replaces https://github.com/tikv/tikv/pull/6644

File: src/server/service/diagnostics/mod.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
```
Comment (BusyJay) [inline src/server/service/diagnostics/mod.rs:1]: Should be 2020.

**Comment by lonng** [comment]
LGTM
@nrc Please resolve the conflicts and address @BusyJay 's comment.

**Comment by crazycs520** [comment]
@nrc Could you cherry-pick this PR to release 4.0?

**Pull request merged**